File: CarHomeWidget/src/main/java/com/anod/car/home/app/CarWidgetActivity.java
package com.anod.car.home.app;

import com.anod.car.home.CarWidgetApplication;
import com.anod.car.home.prefs.preferences.AppTheme;

import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;
import android.support.v7.app.AppCompatActivity;

/**
 * @author alex
 * @date 11/20/13
 */
abstract public class CarWidgetActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        int theme = getApp().getThemeIdx();
        setTheme(getAppThemeRes(theme));
        super.onCreate(savedInstanceState);

    }

    protected int getAppThemeRes(int theme) {
        return (isTransparentAppTheme()) ? AppTheme.getTransparentResource(theme)
                : AppTheme.getMainResource(theme);
    }

    protected boolean isTransparentAppTheme() {
        return false;
    }

    public CarWidgetApplication getApp() {
        return CarWidgetApplication.get(this);
    }

}

File: CarHomeWidget/src/main/java/com/anod/car/home/app/CarWidgetActivity.kt
package com.anod.car.home.app

import com.anod.car.home.CarWidgetApplication
import com.anod.car.home.prefs.model.AppTheme

import android.os.Bundle
import androidx.annotation.StyleRes
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate

/**
 * @author alex
 * @date 11/20/13
 */
abstract class CarWidgetActivity : AppCompatActivity() {

    val app: CarWidgetApplication
        get() = App.get(this)

    val theme: AppTheme
        get() = App.theme(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(appThemeRes)
        AppCompatDelegate.setDefaultNightMode(app.nightMode)
        super.onCreate(savedInstanceState)

    }

    protected open val appThemeRes: Int
        get() = theme.mainResource

}
